tests/post/steps/test_authentication.py

```diff
@@ -271,4 +273,30 @@ def _dex_auth_request(control_plane_ip, username, password):
     return result
 
 
+# a retry helper that performs 3 retries with an exponential sleep interval
+# between each request.
+# Only retry internal server errors and service unavailable errors
+# Source: https://www.peterbe.com/plog/best-practice-with-retries-with-requests
```

Comment:
Let's convert this into a nice docstring, following [Google code style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)

(the link to the article can stay as a comment, not a doctsring as it's more about implementation details)

Also, this method seems generic & useful enough to belong to `tests/utils.py` instead of being hidden here.